esp-hal/src/timer/systimer.rs

```diff
@@ -951,12 +770,22 @@ where
     fn set_interrupt_handler(&self, handler: InterruptHandler) {
         self.comparator.set_interrupt_handler(handler);
     }
+
+    async fn wait(&self) {
+        asynch::AlarmFuture::new(self).await
+    }
+
+    fn async_interrupt_handler(&self) -> InterruptHandler {
```

Comment:
(Long term) Should the interrupt priority be decided down here?

Reply:
I don't think so, it's a bit low-level to do it here

Reply:
Just to be sure you understood what I'm saying, since you resolved the comment but didn't mention anything about a potential future PR.
By returning `InterruptHandler` here you're deciding the priority at this level. You should be returning a function pointer instead, no?

Reply:
The handler can be obtained from `InterruptHandler` and reconstructed with a different priority. We don't have a way in general to set the prio of asynch handlers so I want to punt this down the road.